core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -270,6 +270,14 @@ private static SqlCall transformConvert(SqlValidator validator, SqlCall call) {
               .andThen(SqlTypeTransforms.TO_NULLABLE_ALL),
           OperandTypes.SAME_SAME);
 
+  /** The "NVL2(value, value, value)" function. */
+  @LibraryOperator(libraries = {SPARK})
```

Comment:
why does the first operand have to have the same type as the other two?

Reply:
I borrowed from NVL, it seems like NVL2 doesn't necessarily have to keep the types the same